components/victron/victron.cpp

```diff
@@ -883,15 +883,15 @@ void VictronComponent::handle_value_() {
 
   // "H17"    0.01 kWh   Amount of discharged energy (BMV) / Amount of produced energy (DC monitor)
   if (label_ == "H17") {
-    // Wh
-    this->publish_state_(amount_of_discharged_energy_sensor_, atoi(value_.c_str()) * 10.0f);  // NOLINT(cert-err34-c)
+    // Wh to kWh
+    this->publish_state_(amount_of_discharged_energy_sensor_, atoi(value_.c_str()) / 100.0f);  // NOLINT(cert-err34-c)
```

Comment:
Excuse me!

I totaly agree with you. I created a fork for my own testing and playing round as I'm quite new to coding and trying to learn. But I played inside the main instead of my own fork. And I couldn't find a way to cancel my mistake, so now it's here....

so again. I'm sorry to bother you.

Reply:
No problem! Let me support you and speed up your learning: If you change the resolution of the sensor at the C++ file you have to update the `sensor.py` too:

https://github.com/KinDR007/VictronMPPT-ESPHOME/blob/main/components/victron/sensor.py#L439-L441

The new `unit_of_measurement` should be the const of `kWh` and the `accuracy_decimals` has changed to `2` decimal places.

This change shouldn't be merged officially because it's a breaking change and will annoy people unnecessarily.

Reply:
yes I found that also, but I couldn't figure out how I got the unit's to KILO_WATT_HOURS instead of WATT_HOURS. 
I added the UNIT_KILO_WATT_HOURS to the listing right at the top op the file, but esphome got me a nice error that it couldn't retrieve this unit of measurement.

And indeed I now just changed the yaml of esphome with a lambda filter and changed the unit of measurement there.

kind regards for your help and have a nice day

Reply:
Cya! Please press the "Close" button at the bottom of this page.